hack/copy-frontend-src-to-ossmc.sh

```diff
@@ -0,0 +1,168 @@
+
+# The absolute source and destination directories
+
+ABS_SOURCE_DIR="${SOURCE_REPO}/${SOURCE_DIR}"
+ABS_DEST_DIR="${DEST_REPO}/${DEST_DIR}"
+
+# Validate the dest directories exist
+
+if [ ! -d "${DEST_REPO}" ]; then
+  echo "You must specify a valid destination git repo location: ${DEST_REPO}"
+  exit 1
+fi
+
+if [ ! -d "${ABS_DEST_DIR}" ]; then
+  echo "The destination directory is missing. Make sure it refers to a valid OSSMC git repo location: ${ABS_DEST_DIR}"
+  exit 1
+fi
+
+# Clone the source repo and checkout the desired branch
+
+echo "Cloning source repo [${SOURCE_REPO_URL}] (branch/tag [${SOURCE_REF}]) into [${SOURCE_REPO}]."
+git clone --single-branch --branch ${SOURCE_REF} ${SOURCE_REPO_URL} ${SOURCE_REPO} &> /dev/null
+
+# Sanity check - make sure what we cloned is expected and found on the actual Kiali github repo
+
+if [ ! -d "${ABS_SOURCE_DIR}" ]; then
+  echo "The source directory is missing. Make sure you cloned the correct repo and branch: ${ABS_SOURCE_DIR}"
+  exit 1
+fi
+
+COMMIT_HASH="$(cd ${ABS_SOURCE_DIR} && git rev-parse HEAD)"
+GITHUB_COMMIT_URL="https://github.com/kiali/kiali/tree/${COMMIT_HASH}/${SOURCE_DIR}"
+
+if ! curl --silent --show-error --fail "${GITHUB_COMMIT_URL}" > /dev/null; then
+  echo "The cloned git source does not appear to exist on GitHub."
+  echo "${GITHUB_COMMIT_URL}"
+  echo "Make sure you checked out a valid branch that tracks the actual Kiali repo."
+  exit 1
+fi
+
+# Determine branch name to use for the new commit and prepare the commit message.
+
+DATETIME_NOW="$(date --utc +'%F-%H-%M-%S')"
+DEST_BRANCH="kiali-frontend-update-${DATETIME_NOW}"
+COMMIT_MESSAGE=$(cat <<EOM
+Updating Kiali frontend source from:
+* git ref:    ${SOURCE_REF}
+* git commit: ${COMMIT_HASH}
+* GitHub URL: ${GITHUB_COMMIT_URL}
+EOM
+)
+
+cat <<EOM
+=======
+Copying:
+  ${ABS_SOURCE_DIR}
+to
+  ${ABS_DEST_DIR}
+=======
+${COMMIT_MESSAGE}
+EOM
+
+# 1. Create a new branch in the dev repo so it can be used as
+#    a PR without affecting any existing branches that already
+#    exist in the dest repo. (Note that this new branch is
+#    based off of the current branch that is checked out. The
+#    user must ensure the branch currently checked out in the
+#    dest repo is the desired one.)
+# 2. Copy the files so the dest dir is a duplicate of the source dir.
+# 3. Commit the files to the local repo with a descriptive commit message.
+
+cd ${ABS_DEST_DIR}
+git checkout -b ${DEST_BRANCH}
+rm -rf ${ABS_DEST_DIR}/{*,.[!.]*}
+cp -R ${ABS_SOURCE_DIR}/* ${ABS_DEST_DIR}
+git add ${ABS_DEST_DIR}
+git commit --quiet --signoff -m "${COMMIT_MESSAGE}"
+
+# Completed!
+# Tell the user what needs to be done next.
+# This script does not automatically push the branch to the remote repo.
+# This script will only ever touch local files so as to avoid any possibility
+# of corrupting the remote repo.
+
+GIT_REMOTE="$(cd ${ABS_DEST_DIR} && for r in $(git remote 2>/dev/null); do if [ "$r" != "origin" ]; then echo ${r}; break; fi; done)"
```

Comment:
In this case 'origin' means the upstream repo right? And the script looks for a different one (the local remote branch). This is funny because I named as 'origin' the local remote branch and 'upstream' the kiali remote one. But this is my configuration, maybe I am the only one doing this hehe.

Reply:
Yeah, your setup is probably the reverse of most people :)

`origin` is typically where the main source originated from (hence the name) - so it typically represents the original remote repo (not a fork). And normally you do NOT want to push to the origin repo (in this case directly to the kiali/openshift-servicemesh-plugin repo). You will want to push the branch to your fork and create a PR from that fork branch (which is why the script looks for the first non-origin one - it is assumed that is your fork.)

But this is only for notification purposes anyway. You can execute whatever command you need when you push the PR branch. This just spits out a message for the "typical" user.